solutions/baseline/variables.tf

```diff
@@ -0,0 +1,297 @@
+  }
+}
+
+variable "cloud_monitoring_agent_tags" {
+  type        = list(string)
+  description = "A list of the tags to associate with the metrics that the IBM Cloud Monitoring agent collects. To add the cluster name as a tag, use the `cloud_monitoring_add_cluster_name` variable."
+  default     = []
+  nullable    = false
+}
+
+variable "cloud_monitoring_add_cluster_name" {
+  type        = bool
+  description = "Whether to attach a tag to log messages. Set to `true` to configure the IBM Cloud Monitoring agent to attach a tag that contains the cluster name to all log messages."
+  default     = true
+}
+
+variable "cloud_monitoring_agent_name" {
+  description = "The name of the IBM Cloud Monitoring agent that is used to name the Kubernetes and Helm resources on the cluster. If a prefix input variable is passed, the name of the IBM Cloud Monitoring agent is prefixed to the value in the `<prefix>-<name>` format."
+  type        = string
+  default     = "sysdig-agent"
+}
+
+variable "cloud_monitoring_agent_namespace" {
+  type        = string
+  description = "The namespace to deploy the IBM Cloud Monitoring agent in. Default value: `ibm-observe`."
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "cloud_monitoring_agent_tolerations" {
+  description = "The list of tolerations to apply to the IBM Cloud Monitoring agent. The default operator value `Exists` matches any taint on any node except the master node. [Learn more](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)"
+  type = list(object({
+    key               = optional(string)
+    operator          = optional(string)
+    value             = optional(string)
+    effect            = optional(string)
+    tolerationSeconds = optional(number)
+  }))
+  default = [{
+    operator = "Exists"
+    },
+    {
+      operator : "Exists"
+      effect : "NoSchedule"
+      key : "node-role.kubernetes.io/master"
+  }]
+}
+
+##############################################################################
+# Logs Agents variables
+##############################################################################
+
+variable "logs_agent_enabled" {
+  type        = bool
+  description = "Whether to deploy the Logs agent."
+  default     = true
+}
+
+variable "logs_agent_name" {
+  description = "The name of the Logs agent. The name is used in all Kubernetes and Helm resources in the cluster."
+  type        = string
+  default     = "logs-agent"
+  nullable    = false
+}
+
+variable "logs_agent_namespace" {
+  type        = string
+  description = "The namespace where the Logs agent is deployed. The default value is `ibm-observe`."
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "logs_agent_trusted_profile" {
+  type        = string
+  description = "The IBM Cloud trusted profile ID. Used only when `logs_agent_iam_mode` is set to `TrustedProfile`. The trusted profile must have an IBM Cloud Logs `Sender` role."
+  default     = null
```

Comment:
You should use the tf 1.9 feature to do cross variable validation here. This can't be null if `logs_agent_iam_mode` is set to `TrustedProfile` and `logs_agent_enabled` is true